src/FindEmailAddress.java

```diff
@@ -0,0 +1,362 @@
+*
+* We implemented a multithread method. each url request and email&href check is a threadTask
+*
+*
+*/
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Hashtable;
+import java.net.URI;
+import java.util.concurrent.*;
+
+/*
+*  This is the main class to get the email addresses
+*
+*/
+public class FindEmailAddress {
+
+    private String rootURL;
+    public String requestDomain;
+    public Hashtable<String,String> urlHistory;
+    public Hashtable<String,String> emails;
+    public static int maxUrlCheck = 200; //to avoid too many pages to check we add maximum number of url to check, it can be modified
+    public ExecutorService executor;
+    public Integer tasksNumber;
+
+    public static void main(String[] args)
+    {
+        String requestUrl = null;
+
+        if(args.length >0)
+        {
+            requestUrl =  args[0];
+        }
+        if (requestUrl == null)
+            return;
+
+        FindEmailAddress findEmailAddress = new FindEmailAddress(requestUrl);
+        findEmailAddress.startWebCrawl();
+        findEmailAddress.printout();
+    }
+
+    /*
+    *  @param requestURL is the root url. All the webpage with same domain will be searched for email address
+    */
+    public FindEmailAddress(String requestURL)
+    {
+        this.rootURL = requestURL;
+        if(!requestURL.toLowerCase().startsWith("http"))
+        {
+            rootURL = "http://" + requestURL;
+        }
+        try {
+            URI uri = new URI(rootURL);
+            this.requestDomain= uri.getHost();
+
+        }catch(java.net.URISyntaxException e)
+        {
+            e.printStackTrace();
+        }
+        this.emails = new Hashtable<String,String>();
+        this.urlHistory = new Hashtable<String, String>();
+        this.urlHistory.put(rootURL,"");
+        tasksNumber = 0;
+    }
+
+    /*
+     *  This is the function to do web crawling
+     *  it will create a thread pool
+     *  if all the tasks are done, it will stop.
+     */
+
+    public void startWebCrawl()
+    {
+        if(rootURL == null || rootURL.trim().length()<=0)
+            return;
+
+        executor = Executors.newFixedThreadPool(10);
+        executor.execute(new Handler(rootURL));
+        do{
+            try {
+                Thread.sleep(3000);
+            }catch(Exception e)
+            {
+                e.printStackTrace();
+            }
+        }while(this.tasksNumber >0);
+
+        executor.shutdown();
+    }
+
+/*
+* this inline class is for http request and email check running task
+*/
+    class Handler implements Runnable {
+        String requestURL;
+        Handler(String requestURL) {
+            this.requestURL = requestURL;
+        }
+        public void run() {
+            System.out.println("start search url :" + requestURL);
+            makeHttpReuqest(requestURL);
```

Comment:
there's a typo in this function name ("request" not "reuqest")

Reply:
thanks